### PR TITLE
Issue #1610: Set node menu defaults.

### LINKED ISSUE #1610: [UX] 'Auto Menu Settings' in core and add Pages to the main menu by default
### Weekly update (19th Dec)

https://github.com/backdrop/backdrop/pull/2999 is RTBC, but more feedback/wording changes requested...

----

Drupal has long had this user-nightmare of lost pages. You made one, now where did it go? Yes, the experienced user knows about the content orphanage where all content is listed, but to a new user losing these pages can be traumatic. Backdrop now has this problem too.

If we placed basic pages into the primary navigation menu by default, then first-time users would be able to find their recently created content without needing to visit the content orphanage.

Experienced users could disable this behavior, or simply move pages on a case-by-case basis out of this menu.

## Proposed solution

* On the Page node form, check the box for "create menu item" by default.
* Add 2 per-content type settings: 1) create menu item by default, and 2) which menu.

---

Advocate for this issue: @BWPanda 

----

Original PR by @BWPanda: ~https://github.com/backdrop/backdrop/pull/2903~
Alternate PR by @jenlampton: ~https://github.com/backdrop/backdrop/pull/2951~
New PR by @BWPanda: https://github.com/backdrop/backdrop/pull/2999

### PR DESCRIPTION
**Pull request by ghost**
Fixes https://github.com/backdrop/backdrop-issues/issues/1610
Replaces https://github.com/backdrop/backdrop/pull/2999

### CODE REVIEW

**Comment by jenlampton** [comment]
tests look good, but I broke the dev site :( trying again...

**Comment by jenlampton** [comment]
sandboxme?

File: core/modules/menu/menu.module [hunk 1]
```diff
@@ -765,12 +765,16 @@ function menu_form_node_type_form_alter(&$form, $form_state) {
     '#group' => 'additional_settings',
     '#weight' => 15,
   );
+  $form['menu']['menu_default'] = array(
+    '#type' => 'checkbox',
+    '#title' => t('Add a link into the menu for new content of this type'),
```
Comment (klonos) [inline core/modules/menu/menu.module:770]:
```suggestion
    '#title' => t('Create a menu link for content of this type, by default'),
```

File: core/modules/menu/menu.module [hunk 2]
```diff
@@ -765,12 +765,16 @@ function menu_form_node_type_form_alter(&$form, $form_state) {
...
+    '#default_value' => $node_type->settings['menu_default'],
+  );
   $form['menu']['menu_options'] = array(
     '#type' => 'checkboxes',
-    '#title' => t('Available menus'),
+    '#title' => t('Menus where this content may be placed'),
```
Comment (klonos) [inline core/modules/menu/menu.module:775]:
```suggestion
    '#title' => t('Menus where content of this type may be placed'),
```

**Comment by quicksketch** [comment]
Looks good! Merged into 1.x for 1.15.0.

**Pull request merged**